torch/utils/_pytree.py

```diff
@@ -181,6 +178,28 @@ def _register_pytree_node(
     SUPPORTED_SERIALIZED_TYPES[cls] = serialize_node_def
     SERIALIZED_TYPE_TO_PYTHON_TYPE[serialized_type_name] = cls
 
+    import torch
+
+    if torch._running_with_deploy():
+        warnings.warn("C++ pytree utilities do not work with torch::deploy.")
+        return
+
+    try:
+        from . import _cxx_pytree as cxx
+    except ImportError:
+        pass
+    else:
```

Comment:
It would be nice if there was some way to tell that cxx pytree is available or not. Something like:
```
import torch.utils._cxx_pytree
torch.utils._cxx_pytree.is_available()
```
In this regime, we would probably need to get the cxx_pytree APIs to error if cxx_pytree is not available